ThirdpartyToolchain.cmake

```diff
@@ -81,11 +81,58 @@ macro(build_folly)
       ${folly_BINARY_DIR}/folly/libfollybenchmark${CMAKE_STATIC_LIBRARY_SUFFIX})
   set(FOLLY_LIBRARIES folly)
 endmacro()
-# ===============================END FOLLY================================
+# ================================= END FOLLY ==================================
+
+# ================================== PROTOBUF ==================================
+
+if(DEFINED ENV{VELOX_PROTOBUF_URL})
+  set(PROTOBUF_SOURCE_URL "$ENV{VELOX_PROTOBUF_URL}")
+else()
+  set(VELOX_PROTOBUF_BUILD_VERSION 21.4)
+  string(
+    CONCAT
+      PROTOBUF_SOURCE_URL
+      "https://github.com/protocolbuffers/protobuf/releases/download/"
+      "v${VELOX_PROTOBUF_BUILD_VERSION}/protobuf-all-${VELOX_PROTOBUF_BUILD_VERSION}.tar.gz"
+  )
+  set(VELOX_PROTOBUF_BUILD_SHA256_CHECKSUM
+      6c5e1b0788afba4569aeebb2cfe205cb154aa01deacaba0cd26442f3b761a836)
+endif()
+
+macro(build_protobuf)
+  message(STATUS "Building Protobuf from source")
+
+  FetchContent_Declare(
+    protobuf
+    URL ${PROTOBUF_SOURCE_URL}
+    URL_HASH SHA256=${VELOX_PROTOBUF_BUILD_SHA256_CHECKSUM})
+
+  if(NOT protobuf_POPULATED)
+    # We don't want to build tests.
+    set(protobuf_BUILD_TESTS
+        OFF
+        CACHE BOOL "Disable protobuf tests" FORCE)
+    set(CMAKE_CXX_FLAGS_BKP "${CMAKE_CXX_FLAGS}")
+    set(CMAKE_CXX_FLAGS
+        "-Wno-stringop-overflow -Wno-missing-field-initializers")
+
+    # Fetch the content using previously declared details
+    FetchContent_Populate(protobuf)
+
+    # Set right path to libprotobuf-dev include files.
+    set(Protobuf_INCLUDE_DIR "${protobuf_SOURCE_DIR}/src/")
+    include_directories("${protobuf_SOURCE_DIR}/src/")
+    add_subdirectory(${protobuf_SOURCE_DIR} ${protobuf_BINARY_DIR})
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS_BKP}")
```

Comment:
I was facing the following error:
```
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Protobuf (missing: Protobuf_LIBRARIES) (found suitable
  version "3.21.4", minimum required is "3.21.4")
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.22/Modules/FindProtobuf.cmake:650 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:314 (find_package)


-- Configuring incomplete, errors occurred!
```
can be reproduced with:
```
$ docker-compose build --no-cache ubuntu-cpp
$ docker-compose run --rm ubuntu-cpp
```
was fixed by adding the following line here:
```
    set(Protobuf_LIBRARIES protobuf)
```
With this change I tested building and running tests on Ubuntu 22.04 using protobuf downloaded via FetchContent:
```
100% tests passed, 0 tests failed out of 144
```

Reply:
As this was in the process of getting merged I created a minor fix PR: https://github.com/facebookincubator/velox/pull/2929

Reply:
Sorry I missed this comment, but thanks for the follow up fix!